### ISSUE TITLE
`rumdl fmt` applies rules excluded via `[per-file-ignores]` as a side effect of fixing another rule

### ISSUE THREAD

**Issue by henryiii** [rvben/rumdl#707]
### rumdl version

rumdl 0.2.27

### Installation method

Pre-built binary

### Bug description

`[per-file-ignores]` is honored by `rumdl check` but not fully by `rumdl fmt`. When `fmt` fixes a non-ignored rule in a file, it re-serializes the surrounding construct and applies *ignored* rules to it as a side effect. The change is not reported in the fix summary.

### Steps to reproduce

`slides/deck.md`:

```markdown
# Title

text
- parent
  * child
```

### Expected behavior

This to be ignored by both check and format steps.

### Actual behavior

`rumdl check` — MD004 is correctly suppressed, only the (non-ignored) MD032 is reported:

```
slides/deck.md:4:1: [MD032] List should be preceded by blank line [*]
Issues: Found 1 issues in 1 file
```

`rumdl fmt` — reports fixing only MD032:

```
slides/deck.md:4:1: [MD032] List should be preceded by blank line [fixed]
Fixed: Fixed 1/1 issues in 1 file
```

But the file now contains a second, unreported change — the nested `*` marker was rewritten to `-`,
i.e. MD004 was applied despite being in `per-file-ignores`:

```markdown
# Title

- parent
  - child
```

### Configuration

```toml
[per-file-ignores]
"slides/**/*.md" = ["MD004"]
```

### Operating system

macOS

### Additional context

This looks like it was opened as #502 but then closed by the original author.

Here's a summary (using AI) of how it affected us in practice:

<details>
<summary>:robot: AI summary :robot:</summary>

How the `rumdl fmt` / `per-file-ignores` bug affected us on SIMPLE-Py:

**The goal:** Scope the Marp-conflicting rules (MD001/MD004/MD024/MD025/MD045) out of `slides/**` so the decks would stop fighting the formatter — exactly what LecrisUT asked for.

**What broke:** `per-file-ignores` looked like the clean fix, and `rumdl check` honored it. But the pre-commit hook runs `rumdl-fmt`, and `fmt` ignores those per-file exclusions. So every commit would have silently rewritten `slides/5_04_distros.md`:
- Demoted per-slide `# Title` headings to `##` — this isn't cosmetic, it **changes how Marp renders the slides**.
- Normalized list markers (`*` → `-`).
- These edits weren't even reported in the fix summary ("Fixed 4/4" while making ~19 more line changes).

**Net effect on us:** `per-file-ignores` was unusable for the one thing we needed it for. We fell back to `exclude = ["slides"]`, which drops the decks from rumdl entirely — coarser than ideal (no linting on slides at all), but the only reliable way to keep `fmt` from mangling them.

**What was unaffected:** The MyST content under `content/` lints and formats fine. The two real issues there (bare footnote URL, heading-level jump) were legitimate and got fixed — those had nothing to do with the bug.

So: minor, fully worked around. The cost is that slide decks get no lint coverage until the upstream `fmt` bug is fixed, at which point we could switch back to selective `per-file-ignores` and keep the useful checks (trailing whitespace, MD032, etc.) on slides.

</details>

* Original issue in PR: https://github.com/scikit-build/SIMPLE-Py/pull/7

AI disclosure: also used to generate a MWE.

**Comment by rvben** [comment]
Fixed in v0.2.28. `rumdl fmt` now hands the per-file-ignore-filtered rule set to the fix coordinator on both the diff and fix paths, matching what `rumdl check` and the LSP already did. A rule listed in `[per-file-ignores]` is no longer applied as a side effect of fixing another rule in the same file, and the summary count is accurate. Added a regression test that drives the `fmt` binary on your exact reproduction (MD004 ignored, MD032 fixed). Thanks for the clear repro!
